### PR TITLE
Update Composer to v2 (includes woocommerce-blocks 3.8.0 and woocommerce-admin 1.7.0)

### LINKED ISSUE #28122: Composer v2 Support
Hi!

In my project, I use Composer to require `woocommerce/woocommerce:^3.0 || ^4.0`. Although my constraint allows for WC 4+, it does not get installed by Composer. This is because [`composer/installers:1.9`][] is the first version that supports Composer v2, while WC 4 [requires][1] the _exact_ version of `1.7`. The same thing prevents me from installing WC `3.6`: it requires the _exact_ version of `1.6`.

Why is that? Why does it need the exact version `1.6` or `1.7` instead of `^1.6` or `^1.7`? Would it be possible for you to change that, in order to prevent this very narrow constraint from blocking newer WooCommerce versions with Composer v2?

[1]: https://github.com/woocommerce/woocommerce/blob/master/composer.json#L13
[`composer/installers:1.9`]: https://github.com/composer/installers/releases/tag/v1.9.0

### PR DESCRIPTION
**Pull request by ObliviousHarmony**
### All Submissions

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR updates the version constraint of the `composer/installers` dependency so that it is more permissive. In order to facilitate this we've also had to make the constraint more permissive in WooCommerce Blocks and WooCommerce Admin, so this PR includes updates to those two packages. If you could leave the appropriate merge PR information as a comment here that would be very helpful!

Closes #28122.

### How to test the changes in this Pull Request:

1. Run `composer install` after updating to Composer v2
2. Ensure all tests pass!

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Dev - Updated to Composer v2

### CODE REVIEW

**Comment by budzanowski** [comment]
## WooCommerce blocks version bump

I see that `woocommerce-blocks` package was bumped to `3.8.0` - wonderful. Noting here that this means that besides `3.8.0` we are also shipping `3.7.0` and `3.7.1` for the first time.

### Blocks 3.8.0
- [Release PR](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3384)
- [Testing instructions](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/docs/testing/releases/380.md)
- [Release Post](https://developer.woocommerce.com/2020/11/11/woocommerce-blocks-3-8-release-notes/)

### Blocks 3.7.1
- [Release PR](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3373)
- [Testing instructions](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/docs/testing/releases/371.md)
- [Release Post](https://developer.woocommerce.com/2020/11/05/woocommerce-blocks-3-7-1-release-notes/)

### Blocks 3.7.0
- [Release PR](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3339)
- [Testing instructions](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/docs/testing/releases/370.md)
- [Release Post](https://developer.woocommerce.com/2020/10/29/woocommerce-blocks-3-7-release-notes/)

## Testing

Most of the changes target the feature plugin and not the WooCommerce core package. I have distilled the testing instructions to contain only the bit relevant to this version bump:

#### Standardise & refactor colors scss to align with Gutenberg colors and WooCommerce brand. ([3300](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3300))
Test all blocks in editor/frontend and verify there are no evident issues with its display (layout completely broken or colors that are not correctly assigned). You can copy & paste this gist into the editor to automatically add all blocks. You might need to paste it with Ctrl+Shift+V so styles are removed and the editor can transform it into blocks.

#### Show current selected attributes when re-edit Products by Attribute block. ([3185](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3185))

1. Add a Products by Attribute block.
2. Select some attributes and click on Done.
3. Click on the pencil icon to edit the block again.
4. Confirm your existing attributes are selected by default.

#### Change delimiter used for splitChunks from tilde to hyphens. ([3360](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3360))

This issue is related to the deployment of a site using Beanstalk and it is not directly testable and should not affect functionality. ( Adding for visibility )
1. Complete a build and smoke test block inserter - atomic blocks would be affected if this broke anything ( in Core those blocks are only accessible when editing the All Products block ).
2. Confirm -- usage in built files.

I have smoke tested the PR and it looks like it is working fine.
I am the release lead for WC Blocks 3.8.0 please tag me on any bugs and issues that are related to this version bump.

**Review by budzanowski — DISMISSED** [review]
Looks good from the WC Blocks side.

File: composer.json [hunk 1]
```diff
@@ -10,13 +10,13 @@
...
     "automattic/jetpack-constants": "1.5.0",
-    "composer/installers": "1.7.0",
+    "composer/installers": "~1.7",
     "maxmind-db/reader": "1.6.0",
     "pelago/emogrifier": "3.1.0",
     "psr/container": "1.0.0",
     "woocommerce/action-scheduler": "3.1.6",
-    "woocommerce/woocommerce-admin": "1.6.3",
-    "woocommerce/woocommerce-blocks": "3.6.0",
+    "woocommerce/woocommerce-admin": "1.7.0-beta.1",
```
Comment (timmyc) [inline composer.json:18]: Can we swap this with "1.7.0" please?

Comment (ObliviousHarmony) [reply composer.json:18]: Yep!

**Review by jonathansadowski — DISMISSED** [review]
Updated to Composer 2 and was able to install without issue with this PR 👍. Nice work.

**Review by jonathansadowski — APPROVED** [review]
Verified that code coverage is passing now 👍

**Pull request merged**